include/cantera/numerics/Integrator.h

```diff
@@ -87,6 +87,19 @@ class Integrator
         warn("setLinearSolverType");
     }
 
+    /** Configure how many event/root functions the integrator should monitor
+     *
+     * Callers toggle this to enable/disable root finding on the fly (when
+     * ReactorNet enforces advance limits).
+     * @param nroots Number of root functions
+     */
+    virtual void setRootFunctionCount(size_t nroots) {
+        (void) nroots;
+    }
```

Comment:
A nicer way to annotate this in C++17:
```suggestion
    virtual void setRootFunctionCount([[maybe_unused]] size_t nroots) {}
```